webapp/src/client.ts

```diff
@@ -48,6 +50,11 @@ export default class CallsClient extends EventEmitter {
         this.onDeviceChange = () => {
             this.updateDevices();
         };
+        this.onBeforeUnload = () => {
+            logDebug('unload');
+            this.disconnect();
+        };
+        window.addEventListener('beforeunload', this.onBeforeUnload);
```

Comment:
Technically if you refresh the page (or any other similar action) the client would completely disconnect anyway. This is really about cleaning up before that happens (sending the leave message) to avoid having to timeout connections on the server.